src/domain/community/community/community.service.ts

```diff
@@ -416,12 +418,60 @@ export class CommunityService {
     return community;
   }
 
+  private async validateCommunityPolicyLimits(
+    community: ICommunity,
+    communityPolicyRole: CommunityPolicyRole,
+    role: CommunityRole,
+    action: CommunityContributorsUpdateType
+  ) {
+    const orgMembers = await this.getOrganizationsWithRole(community, role);
```

Comment:
This is going to be very heavy in two dimensions: 
- you only need the count, not the full organization loaded
- this implementation always checks both users and organizations, when there will never be more than one being checked at a time.